### DISCUSSION TITLE
How to make pinentry working when using ssh command?

### DISCUSSION THREAD

**Discussion by amalgame21** [dyne/tomb#467] [Q&A]
I have a tomb file on remote machine.
If I ssh to it, then use tomb command to mount it, everything is fine.

I want to create a script to automate this process and use rsync to back up files.
One of the line in the script is `ssh remote-device 'tomb open /path/to/file.tomb -k /path/to/file.tomb.key'`
However, the output displayed the pinentry is inappropriate.

How can I get around this?
Thanks!

**Comment by Narrat** [comment]
One way around may the use of `sshfs`? Instead of mounting it on the remote machine mount the dir and open the tomb locally?

In any case I will try to take a look at this this weekend. (Or maybe you had in the meantime a different solution?)

**Reply by Narrat** [reply]
And more generally asked. What pinentry program do you expect to see? Or what happens normally when you use it locally?
If you were a user of `pinentry-curses` or `pinentry-tty` there shouldn't be any problem.

Another approach could be to use of the gpg integration in tomb and make use of the `GPG_TTY` variable in the script. This should ignore `DISPLAY`.

**Comment by jaromil** [comment]
Maybe your problem with pinentry is fixed in the latest 2.11 release: I added support for `pinentry-tty` which indeed works better over ssh, but before this fix wasn't used even if installed. Happy to know how goes.
